### PR TITLE
[fr] 3.4 response docs update (follow #4537)

### PR DESCRIPTION
**Pull request by HavokInspiration**
Follows #4537 for the French docs.

I modified some things I think were missed in the original version, be sure to check it out before merging.

### CODE REVIEW

File: fr/controllers/request-response.rst [hunk 1]
```diff
@@ -601,71 +584,78 @@ de caractères::
 Définir les En-têtes
 --------------------
 
-.. php:method:: header($header = null, $value = null)
+.. php:method:: withHeader($header, $value)
+
+Les définitions de headers se font avec la méthode :php:meth:`Cake\\Http\\Response::withHeader()`.
+Comme tous les méthodes de l'interface PSR-7, cette méthode retourne une
```
Comment (cake17) [inline fr/controllers/request-response.rst]: toutes les méthodes

File: fr/controllers/request-response.rst [hunk 2]
```diff
@@ -864,22 +855,35 @@ et ses en-têtes liés avec une interface simple::
...
+un peu de temps. Toutes les méthodes qui commencent par ``with`` intéragiront
+avec la réponse à la manière immutable et retourneront **toujours** une
+**nouvelle** instance. L'erreur la plus fréquente quand les développeurs
+travaillent avec les objets immutables est d'oublier de persister l'instance
+modifiée::
+
+    $this->response->withHeader('X-CakePHP', 'yes!');
+
+Dans le code ci-dessus, la réponse ne contiendra pas le header ``X-CakePHP``
+car la valeur retournée par ``withHeader()`` n'a pas été persisté. Pour avoir
```
Comment (cake17) [inline fr/controllers/request-response.rst]: persistée

File: fr/controllers/request-response.rst [hunk 3]
```diff
@@ -845,7 +836,7 @@ il supprime le contenu de la réponse et envoie l'en-tête `304 Not Modified`::
 
 .. _cors-headers:
 
-Définir les En-têtes de Requête d'Origine Croisée (Cross Origin Request Headers = CORS)
+Définir les En-têtes de Requête d'Origine Croisée (Cross Origin ServerRequest Headers = CORS)
 =======================================================================================
```
Comment (cake17) [inline fr/controllers/request-response.rst]: missing some ====

Comment (HavokInspiration) [inline fr/controllers/request-response.rst]: Indeed and the "ServerRequest" should be "Request". The Search and Replace was a bit destructive.

**Comment by HavokInspiration** [comment]
Thanks for the review @cake17. I'll probably make the necessary changes tomorrow.

**Comment by HavokInspiration** [comment]
@cake17 I took care of it.

**Comment by antograssiot** [comment]
Nice work @HavokInspiration

**Comment by cake17** [comment]
top @HavokInspiration !

**Pull request merged**